File: src/main/kotlin/weatherInfo/javaVersion/WeatherInfo.java
package weatherInfo.javaVersion;

import static weatherInfo.javaVersion.WeatherInfo.Color.*;

public class WeatherInfo {
    private String prompt;
    private Color color;

    public WeatherInfo(int degrees) {
        updateWeatherInfo(degrees);
    }

    public void updateWeatherInfo(int degrees) {
        final String description;
        final Color color;
        if (degrees <= 18) {
            description = "cold";
            color = BLUE;
        } else if (degrees <= 28) {
            description = "mild";
            color = ORANGE;
        } else {
            description = "hot";
            color = RED;
        }
        this.prompt = "The weather is " + description + " now!";
        this.color = color;
    }

    public String getPrompt() {
        return prompt;
    }

    public Color getColor() {
        return color;
    }

    public enum Color {
        BLUE, ORANGE, RED
    }
}

File: src/main/kotlin/weatherInfo/kotlinVersion/WeatherInfo.kt
package weatherInfo.kotlinVersion

import weatherInfo.kotlinVersion.WeatherInfo.Color.*

class WeatherInfo(degrees: Int) {
    var prompt: String? = null
        private set
    var color: Color? = null
        private set

    init {
        updateWeatherInfo(degrees)
    }

    fun updateWeatherInfo(degrees: Int) =
        when {
            degrees <= 18 -> "cold" to BLUE
            degrees <= 28 -> "mild" to ORANGE
            else -> "hot" to RED
        }.run {
            prompt = "The weather is $first now!"
            color = second
        }

    enum class Color {
        BLUE,
        ORANGE,
        RED
    }
}

